### PR TITLE
exclude types.RoleMDM from process.waitForInstanceClient()

### PR DESCRIPTION
**Pull request by flyinghermit**
https://github.com/gravitational/teleport/pull/30384 Introduced changes that would wait for instance client except for process with `types.RoleInstance` role. 

The Jamf service runs with `types.RoleMDM`. When running as a standalone service, it implicitly acquires `types.RoleInstance`
https://github.com/gravitational/teleport/blob/21d1d243d38daf12f692277047ff620a453b53c4/api/types/system_role.go#L63-L70

But when running as Hosted Plugin instance, it does not, so the Jamf plugin hits `process.waitForInstanceClient()` block and times out.  
This PR adds `types.RoleMDM` to exclusion, along with existing `types.RoleInstance`. 

Fixes https://github.com/gravitational/teleport.e/issues/2022

### CODE REVIEW

File: lib/service/connect.go
Comment (sshahcodes) [inline lib/service/connect.go:1050]: This is as quickly I could come up with but lmk if there's more better way to resolve this.

Comment (sshahcodes) [reply lib/service/connect.go:1050]: FYI @codingllama @justinas tagging you both if there's anything else to consider from Jamf service and hosted plugin perspective.

Comment (zmb3) [reply lib/service/connect.go:1050]: Definitely want @fspmarshall to look at this as well.

Comment (codingllama) [reply lib/service/connect.go:1050]: Could someone spell out the consequences of this change?

Would this same condition hit other plugins, or future services that don't need extraordinary permissions?

Would it be correct for plugins to request RoleInstance as well, instead of this?

**Comment by codingllama** [comment]
While we wait for a fix, should we temporarily skip the broken plugin tests so other PRs can land?

**Comment by sshahcodes** [comment]
> While we wait for a fix, should we temporarily skip the broken plugin tests so other PRs can land?

Since the issue breaks Jamf plugin functionality, I would prefer to get the hotfix merged rather than excluding the test itself. We also wont be able to catch similar changes that may be introduced with other PR. BUT, given this is not queued for immediate release, unblocking other PRs is a good idea. UI unit test in `e` is also failing for different reason and the fix(https://github.com/gravitational/teleport.e/pull/2020) is blocked by this issue . 

Any thoughts on temporarily skipping test  @codingllama @zmb3 @rosstimothy @fspmarshall 
Please give thumbs up to this comment if you agree (I'll move ahead with 2 votes 😀)

**Review by fspmarshall — APPROVED** [review]
Approving with the caveat that this should be a temporary fix until we know what the real cause of the problem is.  There isn't any serious downside to letting service-specific certs continue to be issued for specific roles, but it's something we'd like to phase out.

File: lib/service/connect.go
Comment (fspmarshall) [reply lib/service/connect.go:1050]: This is harmless, it just causes allocation of mdm certs to fallback to the pre-https://github.com/gravitational/teleport/pull/30384 behavior.

**Review by codingllama — APPROVED** [review]

**Comment by sshahcodes** [comment]
Thanks for approval. I acknowledge this as a temporary fix and will look into more detail late.

**Pull request merged**
